### PR TITLE
DEVPROD-10198: Show updated task status legend on waterfall page

### PR DESCRIPTION
**Pull request by minnakt**
DEVPROD-10198

### Description
Adds new task status legend while keeping support for old task status legend.

In order for floating content to be positioned properly, I set `overflow-x: hidden` on the waterfall page. I'm not really sure why that works but it fixes the position of the portalled elements during scroll (tooltip & popover) 🤷 I checked Safari and didn't notice any performance issue... Another option is to just set `usePortal` to false, although this has some weird side effects with opacity.

### Screenshots

Order and layout should match Figma designs

Screenshot 2024-11-19 at 3 31 15 PM

### CODE REVIEW

File: apps/spruce/src/components/TaskStatusIconLegend/index.tsx [hunk 1]
```diff
@@ -1,109 +1,152 @@
...
+export const LegendContent: React.FC<LegendContentProps> = ({
+  useWaterfall,
+}) => {
+  const Container = useWaterfall
+    ? WaterfallContainer
+    : MainlineCommitsContainer;
+
+  const groupedStatuses = useWaterfall
+    ? waterfallGroupedStatuses
+    : groupedIconStatuses;
```
Comment (sophstad) [inline apps/spruce/src/components/TaskStatusIconLegend/index.tsx:32]: Hm this is supposed to show the old icon set on mainline commits right? That isn't working for some reason (and I do think showing separate icon sets is the desired behavior)
image

Comment (minnakt) [reply apps/spruce/src/components/TaskStatusIconLegend/index.tsx:33]: added more boolean logic to make this behave better !

File: apps/spruce/src/components/Content/Layout.tsx [hunk 2]
```diff
@@ -55,7 +56,8 @@ export const Layout: React.FC = () => {
         />
       )}
       <FloatingContent>
-        <TaskStatusIconLegend />
+        {/* TODO: Use user's beta feature setting instead. */}
```
Comment (sophstad) [inline apps/spruce/src/components/Content/Layout.tsx:59]: Let's include the ticket number here
```suggestion
        {/* TODO DEVPROD-10203: Use user's beta feature setting instead. */}
```

File: apps/spruce/src/components/TaskStatusIconLegend/index.tsx [hunk 3]
```diff
@@ -1,109 +1,158 @@
...
-                setIsActive(false);
+                setOpen(false);
               }}
             >
               <Icon glyph="X" />
             </IconButton>
           </TitleContainer>
-          <LegendContent />
+          <LegendContent
+            useWaterfall={isWaterfall || (useWaterfall && !isMainlineCommits)}
```
Comment (sophstad) [inline apps/spruce/src/components/TaskStatusIconLegend/index.tsx:140]: If I'm not mistaken the build variant and task history pages use the old icon set. Maybe for now we should only show the new icon set on the waterfall page and default to the old set. When the task history page is redone we can update the legend there too.

**Review by sophstad — CHANGES_REQUESTED** [review]

File: apps/spruce/src/components/TaskStatusIconLegend/index.tsx (hunk 3 above)
Comment (minnakt) [reply apps/spruce/src/components/TaskStatusIconLegend/index.tsx:140]: ok sounds good!

**Review by sophstad — APPROVED** [review]

**Pull request merged**